src/Symfony/Component/DependencyInjection/Dumper/PhpDumper.php

```diff
@@ -652,6 +660,10 @@ private function addService($id, $definition)
         if ($definition->isSynthetic()) {
             $code .= sprintf("        throw new RuntimeException('You have requested a synthetic service (\"%s\"). The DIC does not know how to construct this service.');\n    }\n", $id);
         } else {
+            if ($definition->isDeprecated()) {
+                $code .= sprintf("        @trigger_error('The service %s has been marked as deprecated. You should stop using it.', E_USER_DEPRECATED);\n\n", $id);
```

Comment:
I think it would be great to allow changing the message, or even better: make it mandatory.